authz/sdk_end2end_test.go

```diff
@@ -0,0 +1,164 @@
+/*
+ *
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package authz
```

Comment:
As a total nit: if you put this in the `authz_test` package (same directory, same file, just change this line here), if possible, it will be a better test:

1. Will look exactly like user code (requires you to call `authz.NewStatic`, e.g.).
2. Will not allow any injection/inspection of internals.

If it's not easy to change that, that's fine, but for an e2e test, it's nicer when it's outside the main package you are testing.

Reply:
Thank you for the pointer:) Updated test code.